queue_processors/queue_processor/queue_processor.py

```diff
@@ -72,7 +73,7 @@ def on_message(self, headers, message):
         logger.info("Destination: %s Priority: %s", destination, self._priority)
         # Load the JSON message and header into dictionaries
         try:
-            self._data_dict = json.loads(message)
+            self.message.populate(json.loads(message))
```

Comment:
do we need the json load here? I think ideally we want to remove the JSON dependency from the queue_processors and encapsulate this into the message itself (e.g. only the message needs to know how it can be packed up and sent between queues)

Reply:
I'm finding that doing this causes the `test_end_to_end` tests to fail (e8afd5d).
I tried quickly integrating Message into them (86279bf), but this still caused errors. I think that will require a more thoughtful approach.

It seems removing this json tolerance in queue_processor requires json to be removed from other areas of the code base. I wonder if the best approach would be to keep this for now by reverting to (2c92250) and adding a note to return to it?